src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentTypeFinder.cs

```diff
@@ -74,6 +74,10 @@ internal static class DependentTypeFinder
                     new AsyncLazy<ImmutableArray<(SymbolKey, ProjectId)>>(
                         async c =>
                         {
+                            // If we're the code that is actually computing the symbols, then just 
+                            // take our result and store it in the outer frame.  That way the caller
+                            // doesn't need to incur the cost of deserializing the symbol keys that
+                            // we're create right below this.
```

Comment:
Thank you for the comments here.  I read this in commit order and spent a LONG time looking at that first commit to understand it.